### PR TITLE
Adding error messages to the search bar

### PR DESCRIPTION
**Pull request by lancewf**
Screen Shot 2019-05-23 at 4 13 11 PM

Screen Shot 2019-05-23 at 4 17 26 PM

### :nut_and_bolt: Description

Allowing error messages to be filter on in the Client Runs's search bar. This adds a new field in the node-state called 'error_message' this field contains the error.message field in the original CCR. The best way to use this feature is with regex. If a user has the error of 'file[/failed/file/resource] (insights-test::default line 26) had an error: Chef::Exceptions::EnclosingDirectoryDoesNotExist: Parent directory /failed/file does not exist.', they can search for just a portion of the message. For example search with `*Chef::Exceptions::EnclosingDirectoryDoesNotExist*` or `*/failed/file does not exist*`. 

### :athletic_shoe: Demo Script / Repro Steps

1. `build components/config-mgmt-service && build components/ingest-service && build components/automate-ui`
1. `start_all_services`
1. `send_chef_run_failure_example`
1. Open https://a2-dev.test/client-runs
1. In the search bar select type 'Error'
1. Type 'does not exist' and ensure a suggestion appears
1. Then create a filter for `*Chef::Exceptions::EnclosingDirectoryDoesNotExist*` and ensure that one node is displayed in the node list. 
1. Remove all filters and create a new Error filter with `*/failed/file does not exist*` and ensure that one node is displayed in the node list. 

### :chains: Related Resources

https://github.com/chef/automate/issues/368

### :white_check_mark: Checklist

- [x] Necessary tests added/updated?
- [ ] Necessary docs added/updated?
- [x] Code actually executed?
- [x] Vetting performed (unit tests, lint, etc.)?

### CODE REVIEW

File: .studio/automate-gateway [hunk 1]
```diff
@@ -223,13 +223,16 @@ function send_chef_run_failure_example() {
...
+  local node_name="node_failed_$(($RANDOM % 100))"
+  local rfc_time=$(date --rfc-3339=seconds  -d "$(($RANDOM % 144)) hour ago" | sed 's/ /T/' | sed 's/\+.*/Z/')
 
   local examples_path="/src/components/ingest-service/examples";
   [[ "$1" == "legacy" ]] && endpoint="/events/data-collector" || endpoint="/ingest/events/chef/run"
 
-  # Update the 'id'
-  tmp_ccr_json="$(jq --arg id "$uuid" '.id = $id' <$examples_path/converge-failure-report.json)"
+  # Update the IDs and names
+  tmp_ccr_json="$(jq --arg rfc_time "$rfc_time" --arg run_uuid "$run_uuid" --arg node_name "$node_name" --arg node_uuid "$node_uuid" '.end_time = $rfc_time | .start_time = $rfc_time | .entity_uuid = $node_uuid | .id = $run_uuid | .run_id = $run_uuid | .node_name = $node_name' <$examples_path/converge-failure-report.json)"
```
Comment (lancewf) [inline .studio/automate-gateway]: Updating this command to create a new node each time it is run.

File: components/config-mgmt-service/integration_test/suggestions_test.go [hunk 2]
```diff
@@ -32,7 +32,7 @@ func TestSuggestionsEmptyRequestReturnsError(t *testing.T) {
 
 func TestSuggestionsWithAnInvalidTypeReturnsError(t *testing.T) {
 	ctx := context.Background()
-	cases := []string{"fake", "my-platform", "PolicyName", "error", "something_else"}
+	cases := []string{"fake", "my-platform", "PolicyName", "something_else"}
```
Comment (lancewf) [inline components/config-mgmt-service/integration_test/suggestions_test.go]: The invalid case became a valid case 😄

**Comment by susanev** [comment]
im just curious, what is the difference between `Misspelling` and `Misspelling your name?`

**Comment by lancewf** [comment]
@susanev I was just creating CCRs with random errors. I was just trying to show that the suggestions works for filtering on errors.

File: .studio/automate-gateway [hunk 3]
```diff
@@ -223,13 +223,16 @@ function send_chef_run_failure_example() {
   install_if_missing core/curl curl
   install_if_missing core/jq-static jq
 
-  uuid=$(uuidgen)
+  local run_uuid=$(uuidgen)
+  local node_uuid=$(uuidgen)
+  local node_name="node_failed_$(($RANDOM % 100))"
+  local rfc_time=$(date --rfc-3339=seconds  -d "$(($RANDOM % 144)) hour ago" | sed 's/ /T/' | sed 's/\+.*/Z/')
```
Comment (phiggins) [inline .studio/automate-gateway]: Some of the default formatting probably depends on the ENV, it'd probably be less error prone to use `date`'s formatting options to get what you want directly.

File: components/config-mgmt-service/params/parameters.go [hunk 4]
```diff
@@ -50,6 +50,8 @@ func ConvertParamToNodeRunBackend(parameter string) string {
 		return backend.ChefVersion
 	case "chef_tags":
 		return backend.ChefTags
+	case "error":
+		return "error_message"
```
Comment (phiggins) [inline components/config-mgmt-service/params/parameters.go]: It looks like the rest of these things are defined as constants, should `error_message` get moved over there?

File: components/config-mgmt-service/integration_test/suggestions_test.go (hunk 2 above)
Comment (vjeffrey) [reply components/config-mgmt-service/integration_test/suggestions_test.go]: haha!

**Pull request merged**